index.js

```diff
@@ -37,10 +37,24 @@ function compressPlugin (fastify, opts, next) {
   const supportedEncodings = ['gzip', 'deflate', 'identity']
   if (opts.brotli) {
     compressStream.br = opts.brotli.compressStream
-    supportedEncodings.push('br')
+    supportedEncodings.unshift('br')
   } else if (zlib.createBrotliCompress) {
     compressStream.br = zlib.createBrotliCompress
-    supportedEncodings.push('br')
+    supportedEncodings.unshift('br')
+  }
+
+  if (opts.encodings && opts.encodings.length < 1) {
+    next(new Error('The `encodings` option array must have at least 1 item.'))
+  }
+
+  const encodings = Array.isArray(opts.encodings)
+    ? supportedEncodings
+      .filter(encoding => opts.encodings.includes(encoding))
+      .sort((a, b) => opts.encodings.indexOf(a) - supportedEncodings.indexOf(b))
+    : supportedEncodings
+
+  if (encodings.length < 1) {
+    next(new Error('None of the passed `encodings` were supported — compression not possible.'))
   }
```

Comment:
Can you add a test that triggers this? Passing an not existing encoding would do the trick